File: examples/calculator.py
import sys

def main():
    print("=== Python Demo: CLI Calculator ===")
    print("Supported operators: +  -  *  /  %")
    print("Type 'q' to quit.\n")

    while True:
        try:
            a_str = input("Enter first number: ").strip()
            if a_str == "q":
                print("Goodbye.")
                break
            a = int(a_str)
            
            op = input("Enter operator (+  -  *  /  %): ").strip()
            b = int(input("Enter second number: ").strip())

            if op == '+': res = a + b
            elif op == '-': res = a - b
            elif op == '*': res = a * b
            elif op == '/':
                if b == 0:
                    print("Error: division by zero.\n")
                    continue
                res = a // b
            elif op == '%':
                if b == 0:
                    print("Error: division by zero.\n")
                    continue
                res = a % b
            else:
                print("Error: unknown operator.\n")
                continue

            print(f"Result: {a} {op} {b} = {res}\n")

        except ValueError:
            print("Error: invalid number — only digits (and leading '-') allowed.\n")

if __name__ == "__main__":
    main()

File: examples/calculator.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

console.log("=== Node.js Demo: CLI Calculator ===");
console.log("Supported operators: +  -  *  /  %");
console.log("Type 'q' to quit.\n");

function ask() {
    rl.question("Enter first number: ", (aStr) => {
        aStr = aStr.trim();
        if (aStr === 'q') {
            console.log("Goodbye.");
            rl.close();
            return;
        }

        const a = parseInt(aStr, 10);
        if (isNaN(a)) {
            console.log("Error: invalid number — only digits (and leading '-') allowed.\n");
            return ask();
        }

        rl.question("Enter operator (+  -  *  /  %): ", (op) => {
            op = op.trim();
            rl.question("Enter second number: ", (bStr) => {
                const b = parseInt(bStr.trim(), 10);
                if (isNaN(b)) {
                    console.log("Error: invalid number — only digits (and leading '-') allowed.\n");
                    return ask();
                }

                let res;
                if (op === '+') res = a + b;
                else if (op === '-') res = a - b;
                else if (op === '*') res = a * b;
                else if (op === '/') {
                    if (b === 0) { console.log("Error: division by zero.\n"); return ask(); }
                    res = Math.floor(a / b);
                }
                else if (op === '%') {
                    if (b === 0) { console.log("Error: division by zero.\n"); return ask(); }
                    res = a % b;
                } else {
                    console.log("Error: unknown operator.\n");
                    return ask();
                }

                console.log(`Result: ${a} ${op} ${b} = ${res}\n`);
                ask();
            });
        });
    });
}

ask();
